packages/cli/api-importers/v2-importer-commons/src/converters/ExampleConverter.ts

```diff
@@ -0,0 +1,668 @@
+        isValid: boolean;
+        /**
+         * Whether the example was coerced to the schema type.
+         *
+         * For oneOf and anyOf, we will aim to return a non-coerced valid example if possible,
+         * then fall back to the first valid example otherwise. For instance:
+         *
+         * ```
+         * oneOf:
+         *   - type: string
+         *   - type: number
+         * ```
+         *
+         * example: 42
+         *
+         * will return, for each branch:
+         *
+         * ```
+         * {
+         *   isValid: true,
+         *   coerced: true,  // coerced to string
+         *   validExample: "42",
+         *   errors: []
+         * }, {
+         *   isValid: true,
+         *   coerced: false,  // not coerced
+         *   validExample: 42,
+         *   errors: []
+         * }
+         * ```
+         *
+         * In this case, the oneOf node will thus return the non-coerced example,
+         * treating the example as a number.
+         */
+        coerced: boolean;
+        validExample: unknown;
+        /**
+         * The errors that occurred during conversion.
+         *
+         * We can't collect errors along the way because for oneOf and anyOf, we don't know
+         * which branch will be valid until we have converted all of them.
+         */
+        errors: Error[];
+    }
+}
+
+export class ExampleConverter extends AbstractConverter<AbstractConverterContext<object>, ExampleConverter.Output> {
+    protected readonly MAX_DEPTH = 12;
+    protected readonly EXAMPLE_STRING = "foo";
+    protected readonly EXAMPLE_NUMBER = 42.0;
+    protected readonly EXAMPLE_BOOLEAN = true;
+    protected readonly EXAMPLE_INTEGER = 42;
+
+    private readonly schema: OpenAPIV3_1.SchemaObject | OpenAPIV3_1.ReferenceObject;
+    private readonly example: unknown;
+    private readonly depth: number;
+    private readonly schemaId?: string;
+
+    constructor({ breadcrumbs, context, schema, example, schemaId, depth = 0 }: ExampleConverter.Args) {
+        super({ breadcrumbs, context });
+        this.example = example;
+        this.schema = schema;
+        this.depth = depth;
+        this.schemaId = schemaId;
+    }
+
+    public async convert({
+        context,
+        errorCollector
+    }: {
+        context: AbstractConverterContext<object>;
+        errorCollector: ErrorCollector;
+    }): Promise<ExampleConverter.Output> {
+        if (this.depth > this.MAX_DEPTH) {
+            return { isValid: true, coerced: false, validExample: this.example, errors: [] };
+        }
+        let resolvedSchema: OpenAPIV3_1.SchemaObject | undefined;
+        if (context.isReferenceObject(this.schema)) {
+            const resolved = await context.resolveReference<OpenAPIV3_1.SchemaObject>(this.schema);
+            if (resolved.resolved) {
+                resolvedSchema = resolved.value;
+            }
+        } else {
+            resolvedSchema = this.schema;
+        }
```

Comment:
nit: Anytime I see a `let` I think "this is a good place for a helper function". That way we can just call something like the following and avoid the `else` entirely with an early `return`.

```typescript
const resolvedSchema = this.resolveSchema(this.schema);
```

```typescript
private async resolveSchema(schema: <>): OpenAPIV3_1.SchemaObject | undefined {
  if (context.isReferenceObject(this.schema)) {
    const resolved = await context.resolveReference<OpenAPIV3_1.SchemaObject>(this.schema);
    if (resolved.resolved) {
      return resolved.value;
    }
    return undefined;
  }
  return this.schema;
}
```